encoder.go

```diff
@@ -130,6 +133,35 @@ func (d *textDecoder) Decode(t string, value []byte) (driver.Value, error) {
 	case "String":
 		return unescape(unquote(v)), nil
 	}
+
+	// got zoned datetime
+	if strings.HasPrefix(t, "DateTime") {
+		timeZoneName := timeZoneRegexp.FindString(t)
+		if timeZoneName == "" {
+			return nil, fmt.Errorf("time zone not found")
+		}
+		var (
+			loc *time.Location
+			err error
+		)
+
+		if d.useDBLocation {
+			timeZoneName = timeZoneName[2 : len(timeZoneName)-2] // remove \' in the beginning and in the end
+			loc, err = time.LoadLocation(timeZoneName)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			loc = d.location
+		}
+
+		var t time.Time
+		if t, err = time.ParseInLocation(timeFormat, unquote(v), loc); err != nil {
+			return t, err
+		}
+		return t.In(d.location), nil
```

Comment:
I believe `ParseInLocation` returns time in proper location, and it does not need to call method `In`

Reply:
It seems like when `loc` is location received from database, it might differ from the one set in driver (`d.location`). So I thought in this case dateTime should be converted to the location set in driver.

Reply:
Ah, Your are right. ok.